### DISCUSSION TITLE
Entities configuration integrations Config Keys

### DISCUSSION THREAD

**Discussion by zigt1301** [mKeRix/room-assistant#342] [Q&A] [answered]
Small question. I'm trying to configure my XiaomiMi sensor and would like to read the [Rolling Average](https://www.room-assistant.io/guide/entities.html#rolling-average) but am not aware of the config key that is needed to do so. Right now I'm calling the entity in the following way in Home Assistant Add-on. But I'm not sure whether it is working. The Config Key for the entity I'm using right now is "xiaomimi_rass_bedroom:"
<img width="263" alt="Screenshot 2020-11-12 at 13 54 27" src="https://user-images.githubusercontent.com/15857773/98942703-ba384b80-24ee-11eb-9c40-ef74267aadf8.png">
Here you can see what the name and address is of the sensor: the name is "RASS Bedroom"
<img width="278" alt="Screenshot 2020-11-12 at 13 58 07" src="https://user-images.githubusercontent.com/15857773/98943008-27e47780-24ef-11eb-8740-597f9c5a2b26.png">

So my question is if anyone knows how to configure XiaomiMi entities? And maybe it would be a good idea to show that information on the [Xiaomi Mi Sensors](https://www.room-assistant.io/integrations/xiaomi-mi.html) webpage and other integration pages.

**Comment by zigt1301** [comment]
Another question that I have about the Entity registry. Can I have the entity registry on one Room-Assistant instance or do all my instances have to have the Entity registry, all having the same data in the registry?

**Reply by mKeRix** [reply]
Each room-assistant instance keeps its own entity registry, as the instances may have different entities depending on what integrations you configure and so on. Distributed entities (such as the ones created by the Bluetooth integrations) are also kept per instance, but will have the same ID on all instances.

**Comment by mKeRix** [comment, marked as answer]
The keys you need to use for the entity behaviors relate to the room-assistant internal IDs of the entities. Unfortunately there is no easy way to view those yet (something that I want to explore in the future), so for now the best way to do this is either running room-assistant in verbose mode and looking at the logs or looking at the entities API by going to `http://<room-assistant-ip>:6415/entities` and looking for the ID of the entity you want to modify.

**Reply by zigt1301** [reply]
Thank you for your response. I had not read the API section of the guide that closely but I have to say I did not understand that the API could be used for finding the IDs. 

> Unfortunately there is no easy way to view those yet, so for now the best way to do this is by looking at the entities API

The API seems like a easy enough way to view the entities!

What I would suggest is adding a line to the Entities section of the Guide describing what you just posted. Maybe something like "You can find your entities IDs by either running room-assistant in verbose mode and looking at the logs or looking at the API by going to `http://<room-assistant-ip>:6415/entities` and looking for the ID of the entity you want to modify."

And adding a similar line to the API section. "You can find all sorts of data about your instances and entities by looking at the entities API by going to `http://<room-assistant-ip>:6415/entities`."

Hope this helps.
